jerry-core/parser/regexp/re-compiler.c

```diff
@@ -388,7 +388,7 @@ re_parse_alternative (re_compiler_ctx_t *re_ctx_p, /**< RegExp compiler context
 
         ECMA_TRY_CATCH (empty,
                         re_parse_char_class (re_ctx_p->parser_ctx_p,
-                                             re_append_char_class,
+                                             (re_char_class_callback) re_append_char_class,
```

Comment:
In my opinion, it is bad practice to cast function pointers. Instead, change `re_append_char_class` to match `re_char_class_callback`'s type and cast the uint32_t to ecma_char_t inside the function.